docs/fuzz-settings.md

```diff
@@ -32,3 +32,56 @@ Base64: 4+Pj4+Pj4+Pj4wo=
 ==96284== ERROR: libFuzzer: timeout after 10 seconds
 SUMMARY: libFuzzer: timeout
 ```
+
+## Custom Hooks
+
+At low level, custom hooks in Jazzer.js allow the user to
+
+- replace any named library functions with their own (hooking nameless functions
+  is not supported),
+- call the original function when needed,
+- read from and write to the original function's arguments,
+- replace the return value of the original function.
+
+Custom hooks are useful for writing bug detectors, removing fuzzing blockers,
+and improving the fuzzing process by providing feedback to the fuzzer.
+
+### Enabling Custom Hooks in the package.json
+
+To enable custom hooks in Jazzer.js, add
+`-h <path_to_file_with_custom_hooks>.js` to the project configuration in
+package.json:
+
+```json
+"scripts": {
+	"fuzz": "jazzer fuzz ... -h <path_to_file_with_custom_hooks>.js"
+}
+```
+
+Several files with custom hooks can be added like this:
+`-h file1.js -h file2.js`. Each of these files can contain definitions of
+multiple hooks.
+
+### Defining Custom Hooks
+
+The file defining custom hooks should import the function `registerReplaceHook`:
+
+```javascript
+const { registerReplaceHook } = require("@jazzer.js/hooking");
+```
+
+To register a custom hook, the function `registerReplaceHook` can be used:
+
+```typescript
+function registerReplaceHook(
+	target: string, // target function name that we want to hook
```

Comment:
We should also explain that the target name for nested "structure" is constructed by concatenating the individual names using dots, like `SomeClass.someMethod.someInnerFunction`.